policy-controller/src/admission.rs

```diff
@@ -317,11 +317,18 @@ impl Validate<ServerSpec> for Admission {
             .list(&kube::api::ListParams::default())
             .await?;
         for server in servers.items.into_iter() {
-            if server.name_unchecked() != name
+            let server_name = server.name_unchecked();
+            if server_name == name
```

Comment:
Whether the server name is the same or not, we care about making sure there is not already a Server that has the same port and overlaps, right? So maybe we should get rid of the Server name check, and just check `server.spec.port` and `Self::overlaps`.

Reply:
I think the original logic here was correct and was to avoid checking a Server against itself.  If `server_name == name` then `server` is the very server that we're trying to create so we shouldn't consider it a conflict if it overlaps with itself.